templates/lobby.html

```diff
@@ -649,22 +649,30 @@
         color = hexToRgb(color);
         color[3] = 255; //alpha channel
 
-        const circleMap = generateCircleMap(Math.floor(lineWidth / 2));
-        const offset = Math.floor(circleMap.length / 2);
-        const imageData = context.getImageData(0, 0, context.canvas.width, context.canvas.height);
 
+        let radius = Math.floor(lineWidth / 2);
+        // calculate bounding box
+        let left = Math.max(0, Math.min(x1, x2) - lineWidth);
+        let top = Math.max(0, Math.min(y1, y2) - lineWidth);
+        let right = Math.max(0, Math.min(context.canvas.width, Math.max(x1, x2) + lineWidth));
+        let bottom = Math.max(0, Math.min(context.canvas.height, Math.max(y1, y2) + lineWidth));
+
+        const imageData = context.getImageData(left, top, right - left, bottom - top);
```

Comment:
Throws when hitting F5: TypeError: Argument 1 of CanvasRenderingContext2D.getImageData is not a finite floating-point value.